### PR TITLE
Add DataDog initialization (for internal testing url only)

### PR DESCRIPTION
**Pull request by daria-github**
This PR initializes DataDog for our internal testing url, and does not impact or track external production users using xmtp.chat. Its sole purpose is to monitor for performance regressions internally before they hit production.

### CODE REVIEW

File: components/App.tsx [hunk 1]
```diff
@@ -1,14 +1,39 @@
...
+    /* The initialization below will only happen 
+    on our internal testing url (alpha.xmtp.chat)
+    and is used for performance testing purposes.
+    This tracking is meant to help surface insights 
+    about performance based on team usage, and flag
+    any performance degradations before they hit our
+    production users. */
+    if (window.location.hostname.includes(ENVIRONMENT.ALPHA)) {
+      datadogRum.init({
+        applicationId: process.env.NEXT_PUBLIC_DATA_DOG_ID ?? "",
```
Comment (neekolas) [inline components/App.tsx:21]: Shouldn't it just bail out with a warning if these aren't set? I can't imagine it actually working with `""`

File: components/App.tsx [hunk 2]
```diff
@@ -1,14 +1,39 @@
...
-  }
+  useEffect(() => {
+    /* The initialization below will only happen 
+    on our internal testing url (alpha.xmtp.chat)
+    and is used for performance testing purposes.
+    This tracking is meant to help surface insights 
+    about performance based on team usage, and flag
+    any performance degradations before they hit our
+    production users. */
+    if (window.location.hostname.includes(ENVIRONMENT.ALPHA)) {
```
Comment (neekolas) [inline components/App.tsx:19]: We don't want this on preview branches?

File: components/App.tsx (hunk 1 above)
Comment (daria-github) [reply components/App.tsx:21]: it wouldn't work in those cases, was just trying to get around a ts error.

File: components/App.tsx (hunk 2 above)
Comment (daria-github) [reply components/App.tsx:19]: We could add it but I don't think it's necessary/this url should cover most of our internal use cases. IMO it's best to keep this to 1 place, so it's clear this is purely for internal use.

**Review by rygine — APPROVED** [review]

**Review by bhavya2611 — APPROVED** [review]
LG

**Pull request merged**